contrib/tox.ini

```diff
@@ -0,0 +1,47 @@
+[tox]
+package_root = .
+min_version = 4.3.3
+envlist =
+    py{37,38,39,310,311}-qa
+skip_missing_interpreters = true
+ignore_base_python_conflict = true
+labels =
+    mypy = py{37,38,39,310,311}-mypy
+    quality = py{37,38,39,310,311}-quality
+    test = py{37,38,39,310,311}-test
+
+
+[testenv]
+runner = ignore_env_name_mismatch
+package = sdist
+allowlist_externals =
+    make
+envname =
+    py37: py37
+    py38: py38
+    py39: py39
+    py310: py310
+    py311: py311
+envdir =
+    py37: {toxinidir}/.tox/py37
+    py38: {toxinidir}/.tox/py38
+    py39: {toxinidir}/.tox/py39
+    py310: {toxinidir}/.tox/py310
+    py311: {toxinidir}/.tox/py311
```

Comment:
Storing `.tox` in `contrib/` feels a little weird, but it means the environments are ignored by `flake8` and part of `.gitignore` for free, so, I'm personally okay with it :shrug: